subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/filter/TestSelectionMatcher.java

```diff
@@ -62,7 +62,17 @@ public boolean matchesTest(String className, String methodName) {
             && matchesPattern(commandLineIncludePatterns, className, methodName);
     }
 
-    private boolean matchesPattern(List<Pattern> includePatterns, String className, String methodName) {
+    private boolean matchesPattern(List<Pattern> includePatterns, String fullQualifiedName, String methodName) {
+        String simpleName = StringUtils.substringAfterLast(fullQualifiedName, ".");
```

Comment:
❌ This doesn't seem right - We don't want to match patterns when only a simple name was given. We want to match the class and method name exactly in that case.